plugins/parsers/registry.go

```diff
@@ -129,11 +131,13 @@ func NewParser(config *Config) (Parser, error) {
 func NewJSONParser(
 	metricName string,
 	tagKeys []string,
+	fieldKeys []string,
```

Comment:
Generally we want to avoid changing exported function signatures, as it destabilizes packages. Perhaps what we could do is add a `newJSONParser` that takes different (or functional) parameters. We could then update those things that call `NewJSONParser` directly to call `NewParser`, which will in turn call `newJSONParser`.